Tests/AWSSDKSwiftTests/test.swift

```diff
@@ -0,0 +1,25 @@
+//
+// test.swift
+// written by Adam Fowler
+// helper functions for tests
+//
+import XCTest
+import NIO
+import AWSSDKSwiftCore
+
+func attempt(function : () throws -> ()) {
+    do {
+        try function()
+    } catch let error as AWSErrorType {
```

Comment:
I'm not sure, but I think you can drop the `let error as`

```
} catch AWSErrorType {

}
```
should be sufficient.

Reply:
`catch AWSErrorType` doesn't compile as `AWSErrorType` is an error type. You would need `catch is AWSErrorType` but that doesn't work in this situation because the variable `error` is not setup in this situation. So you do need the `let error as` unless there is another way to do this.

Reply:
you're right